### DISCUSSION TITLE
Refresh terminal background color

### DISCUSSION THREAD

**Discussion by aorith** [tmux/tmux#3994] [General]
So with OSC11 we can query the background color of the terminal:

```
❯ printf '\e]11;?\007'
^[]11;rgb:1818/1818/1818^G
```

Some terminals allow a live reload of their configuration, I'm using alacritty with an script to switch from a dark to a light theme.

The issue is that tmux doesn't refresh the terminal background color and the OSC11 sequence returns the old background color causing TUI apps to think that I'm still using a dark theme. If I close all tmux sessions and re-open tmux then it detects the new background.

Is there a way to tell tmux to query again the terminal background color?

**Comment by rvighne** [comment]
You don't need to go so far as to close the tmux sessions, just detach and reattach is enough to refresh tmux's notion of the background color (at least as of 2.6a). However, it would be a much better experience to not have to do that either. Could tmux fetch this value every time it receives the request itself, essentially just forwarding it to the real tty?
